Extend/Order_Controller.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Extend;
+
+use \OxidEsales\Eshop\Core\Registry;
+use \Wirecard\Oxid\Extend\Payment_Gateway;
+use \Wirecard\Oxid\Extend\Order;
+use \Wirecard\Oxid\Core\Payment_Method_Factory;
+use \Wirecard\PaymentSdk\Entity\Amount;
+use \Wirecard\PaymentSdk\TransactionService;
+use \Wirecard\PaymentSdk\Response\FailureResponse;
+use \Wirecard\PaymentSdk\Response\InteractionResponse;
+use \Wirecard\PaymentSdk\Entity\Redirect;
+
+/**
+ * Class Order
+ *
+ * @package Wirecard\Extend
+ *
+ * @mixin \OxidEsales\Eshop\Application\Model\Order
+ */
+class Order_Controller extends Order_Controller_parent
+{
+    /**
+     * Extends the parent init function and finalizes the order in case it
+     * was a Wirecard payment method
+     */
+    public function init()
+    {
+        $oLogger = Registry::getLogger();
+        parent::init();
+
+        $oConfig = Registry::getConfig();
+        $wdPaymentRedirectBack = $oConfig->getRequestParameter('wdpayment');
+        $session = Registry::getSession();
+        $sWdSessionToken = $session->getVariable('wdtoken');
+
+        if (!empty($sWdSessionToken) && $sWdSessionToken == $wdPaymentRedirectBack) {
+            $shopBaseURL = Registry::getConfig()->getShopUrl();
+            $languageCode = Registry::getLang()->getBaseLanguage();
+
+            $aParams = array(
+                'lang' => $languageCode,
+                'force_sid' => $session->getId(),
+                'stoken' => $session->getSessionChallengeToken(),
+                'actcontrol' => 'order',
+                'cl' => 'order',
+                'fnc' => 'execute',
+                'sDeliveryAddressMD5' => $this->getDeliveryAddressMD5(),
+                'challenge' => '',
+                'ord_agb' => '1',
+                'oxdownloadableproductsagreement' => '0',
+                'oxserviceproductsagreement' => '0',
+                'wdtoken' => $sWdSessionToken
+            );
+
+            $sParamStr = '';
```

Comment:
You should use [http_build_query](http://php.net/manual/en/function.http-build-query.php) for this, because it also encodes the URL properly.